### DISCUSSION TITLE
How to explain the output of LabelPropagation model?

### DISCUSSION THREAD

**Discussion by kenenbek** [pyg-team/pytorch_geometric#8283] [Show and tell]
I have a simple setting, where I have two nodes, and an edge goes from node-0 to node-1. 
node-0 has class 0 and  node-1 has class 1

```python
from torch_geometric.nn.models import LabelPropagation
import torch

y = torch.tensor([0, 1])
idx = torch.tensor(([[0, 1]])).T

model = LabelPropagation(num_layers=1, alpha=1)
out = model(y=y, edge_index=idx, mask=[0]) # The same with mask=[0, 1]
```
```
out
tensor([[0., 0.],
        [0., 0.]])
```

I would expect that such setting with alpha=1 will just interchange labels of nodes. And `out` will be like this:

```
out
tensor([[1., 0.],
        [0., 1.]])
```

But I get tensor of zeros. How to explain it?

**Comment by rusty1s** [comment]
Note that you need to define a bidirectional edge:
```python
import torch

from torch_geometric.nn.models import LabelPropagation

y = torch.tensor([0, 1])
edge_index = torch.tensor([[0, 1], [1, 0]])

model = LabelPropagation(num_layers=1, alpha=1)
out = model(y=y, edge_index=edge_index)
print(out)
